### DISCUSSION TITLE
USB/UART converter for programming - 3,3V power problem with NSPanel

### DISCUSSION THREAD

**Discussion by proasnet** [Blackymas/NSPanel_HA_Blueprint#2064] [Q&A]
Dear all,
I tried load esphome to NSPanel, with two different usb/uart converters, but was problem with 3,3V current. After connect boart to converter, my laptop sound disconnect a converter. I think the supply current has been exceeded. I normally program esp chips with both converters. But there is also a display in the NSPanel and it probably increases consumption, so much so that it completely disables the converter chip.
So I have 2 options.
I will either add a 3.3V power supply
or please recommend a converter that has a strong output power current

**Comment by edwardtfn** [comment]
Are you sure your are wiring this correctly? And are you sure your wires are not touching the metal plate in the back?
I've tried different cheap adapters and never faced any issue with this. And I'm pretty sure the screen doesn't turn on when powered via ttl adapter with 3.3V.

**Comment by proasnet** [comment]
I definitely have the wiring correct. I have already programmed ESP a lot. But yes, on the very first connection I went through the pins all the way to the metal plate for a split second. But then I was just adding the pins with an angle, not through the board. That didn't help either. I thought I had destroyed the converter by shorting it to the metal plate, so I took a second one, but that didn't help. I assembled the NSPanel and tested it with the original sonoff sw to see if it is ok and it works.
For both converters that I have, when I attach the pins to the nspanel board, the driver in windows immediately exits, and after disconnecting, the driver starts again. I know it by the sound in windows.

**Comment by edwardtfn** [comment]
So, the one I've used mostly (and which are on the pics for the [installantion guide](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/install.md#connecting-your-device-via-usb-to-serial-ttl-interface) is this one:

- https://www.amazon.se/-/en/dp/B08LMHWK84/

It always worked fine with me.

**Comment by proasnet** [comment]
I've already done it. I fed the wires directly to the pins, without using a comb.
